### PR TITLE
split_controller: reorganize the structs and add more comments

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### What is changed and how it works?

What's Changed:

```commit-message
Reorganize the structs and add more comments.
```

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
None.
```

### CODE REVIEW

**Review by lhy1024 — APPROVED** [review]

**Review by rleungx — APPROVED** [review]

File: components/raftstore/src/store/worker/split_controller.rs [hunk 1]
```diff
@@ -99,57 +75,23 @@ where
...
-#[derive(Debug, Clone)]
-pub struct RegionInfo {
-    pub sample_num: usize,
-    pub query_stats: QueryStats,
-    pub peer: Peer,
-    pub key_ranges: Vec<KeyRange>,
-    pub flow: FlowStatistics,
+pub struct Sample {
+    pub key: Vec<u8>,
+    // left means the number of key ranges located in the sample's left.
```
Comment (AndreMouche) [inline components/raftstore/src/store/worker/split_controller.rs:80]:
```suggestion
    // left means the number of key ranges located on the sample's left.
```

File: components/raftstore/src/store/worker/split_controller.rs [hunk 2]
```diff
@@ -99,57 +75,23 @@ where
...
+    pub left: i32,
+    // contained means the number of key ranges the sample locates inside.
+    pub contained: i32,
+    // right means the number of key ranges located in the sample's right.
```
Comment (AndreMouche) [inline components/raftstore/src/store/worker/split_controller.rs:84]:
```suggestion
    // right means the number of key ranges located on the sample's right.
```

**Review by AndreMouche — APPROVED** [review]
Rest LGTM

**Pull request merged**